src/platform/w3dengine/client/w3dbridgebuffer.cpp

```diff
@@ -0,0 +1,790 @@
+ *            A full copy of the GNU General Public License can be found in
+ *            LICENSE
+ */
+#include "w3dbridgebuffer.h"
+#include "assetmgr.h"
+#include "baseheightmap.h"
+#include "dx8indexbuffer.h"
+#include "dx8vertexbuffer.h"
+#include "dx8wrapper.h"
+#include "globaldata.h"
+#include "mapobject.h"
+#include "mesh.h"
+#include "meshmdl.h"
+#include "shadermanager.h"
+#include "terrainroads.h"
+#include "texture.h"
+#include "w3dshroud.h"
+#include "w3dterrainlogic.h"
+#include <cstring>
+using std::strcat;
+using std::strcpy;
+
+ShaderClass g_detailAlphaShader(361659);
+
+W3DBridge::W3DBridge() :
+    m_scale(1.0f),
+    m_bridgeTexture(nullptr),
+    m_leftMesh(nullptr),
+    m_sectionMesh(nullptr),
+    m_rightMesh(nullptr),
+    m_visible(false),
+    m_curDamageState(BODY_PRISTINE)
+{
+}
+
+W3DBridge::~W3DBridge()
+{
+    Clear_Bridge();
+}
+
+void W3DBridge::Clear_Bridge()
+{
+    m_visible = false;
+    Ref_Ptr_Release(m_bridgeTexture);
+    Ref_Ptr_Release(m_leftMesh);
+    Ref_Ptr_Release(m_sectionMesh);
+    Ref_Ptr_Release(m_rightMesh);
+}
+
+bool W3DBridge::Cull_Bridge(CameraClass *camera)
+{
+    bool visible = m_visible;
+    m_visible = true;
+    return visible != true;
+}
+
+void W3DBridge::Get_Bridge_Info(BridgeInfo *info)
+{
+    info->from.x = m_start.X;
+    info->from.y = m_start.Y;
+    info->from.z = m_start.Z;
+    info->to.x = m_end.X;
+    info->to.y = m_end.Y;
+    info->to.z = m_end.Z;
+    info->bridge_width = (m_maxY - m_minY) * m_scale;
+
+    Vector3 normal((m_end - m_start).X, -(m_end - m_start).Y, 0.0f);
+    normal.Normalize();
+
+    info->from_left.x = normal.X * m_maxY * m_scale + m_start.X;
+    info->from_left.y = normal.Y * m_maxY * m_scale + m_start.Y;
+    info->from_left.z = normal.Z * m_maxY * m_scale + m_start.Z;
+    info->from_right.x = normal.X * m_minY * m_scale + m_start.X;
+    info->from_right.y = normal.Y * m_minY * m_scale + m_start.Y;
+    info->from_right.z = normal.Z * m_minY * m_scale + m_start.Z;
+    info->to_left.x = normal.X * m_maxY * m_scale + m_end.X;
+    info->to_left.y = normal.Y * m_maxY * m_scale + m_end.Y;
+    info->to_left.z = normal.Z * m_maxY * m_scale + m_end.Z;
+    info->to_right.x = normal.X * m_minY * m_scale + m_end.X;
+    info->to_right.y = normal.Y * m_minY * m_scale + m_end.Y;
+    info->to_right.z = normal.Z * m_minY * m_scale + m_end.Z;
+}
+
+void W3DBridge::Init(Vector3 from_loc, Vector3 to_loc, Utf8String bridge_template_name)
+{
+    m_start = from_loc;
+    m_end = to_loc;
+    m_template = bridge_template_name;
+    m_enabled = true;
+}
+
+bool W3DBridge::Load(BodyDamageType cur_damage_state)
+{
+    Ref_Ptr_Release(m_bridgeTexture);
+    Ref_Ptr_Release(m_leftMesh);
+    Ref_Ptr_Release(m_sectionMesh);
+    Ref_Ptr_Release(m_rightMesh);
+    char texture[260];
+    char model[260];
+    char left[260];
+    char section[260];
+    char right[260];
+    memset(texture, 0, sizeof(texture));
+    memset(model, 0, sizeof(model));

```

Comment:
you can just do `={};` to memset them to zero and I suggest doing that for all of the strings.

Reply:
fixed.